lib/spring/client/binstub.rb

```diff
@@ -51,6 +50,7 @@ class Binstub < Command
       BINSTUB_VARIATIONS = Regexp.union [
         %{begin\n  load File.expand_path("../spring", __FILE__)\nrescue LoadError\nend\n},
         %{begin\n  load File.expand_path('../spring', __FILE__)\nrescue LoadError\nend\n},
+        %{begin\n  spring_bin_path = File.expand_path('../spring', __FILE__)\n  load spring_bin_path\nrescue LoadError => e\n  raise unless e.message.end_with? spring_bin_path, 'spring/binstub'\nend\n},
```

Comment:
Well it's much easier now that the variations are listed and used by both upgrade and remove.